dev/release/release-build.sh

```diff
@@ -225,9 +233,21 @@ function checkout_code {
     rm -rf $RELEASE_WORK_DIR
     mkdir $RELEASE_WORK_DIR
     cd $RELEASE_WORK_DIR
-    git clone https://git-wip-us.apache.org/repos/asf/incubator-systemml.git
+    if [[ "$GIT_BRANCH" ]]; then
+        git clone -b $GIT_BRANCH https://git-wip-us.apache.org/repos/asf/incubator-systemml.git
```

Comment:
since you are doing the `git checkout $GIT_BRANCH` below, you should not need to do a `clone -b $GIT_BRANCH ...` here